src/FromData.purs

```diff
@@ -61,13 +85,20 @@ instance FromData ByteArray where
   fromData (Bytes res) = Just res
   fromData _ = Nothing
 
+-- Nothing prevents fromData b ~ Maybe BigInt from being zero here, perhaps
+-- we want more safety:
 instance (Ord a, EuclideanRing a, FromData a) => FromData (Ratio a) where
   fromData (List [ a, b ]) = reduce <$> fromData a <*> fromData b
   fromData _ = Nothing
 
 instance FromData PlutusData where
   fromData = Just
 
+-- | This covers `Bech32` which is just a type alias for `String`
+instance FromData String where
```

Comment:
I'm against this, the user might expect it to work with any string. We only have bytestring instance in plutus as far as I remember, let's keep it that way.

Reply:
That's fair, I was a bit unsure on this TBH. This will affect many other things including `PaymentPubKey` but not `PaymentPubKeyHash` which *I think* is the important one.

Reply:
Removed BTW.